keanu-project/src/main/java/io/improbable/keanu/vertices/SimpleVertexDictionary.java

```diff
@@ -0,0 +1,30 @@
+package io.improbable.keanu.vertices;
+
+import java.util.Arrays;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import io.improbable.keanu.tensor.Tensor;
+
+public class SimpleVertexDictionary implements VertexDictionary {
+
+    private final Map<VertexLabel, Vertex<?>> map;
+
+    private SimpleVertexDictionary(Map<VertexLabel, Vertex<?>> map) {
+        this.map = map;
+    }
+
+    @Override
+    public <V extends Vertex<? extends Tensor<?>>> V get(VertexLabel label) {
```

Comment:
Why does this have to be a tensor vertex?

Reply:
it doesn't - changed.

I think at some intermediate point in my coding I needed this, but now I don't (fortunately).

Reply:
It's the issue that I emailed around the other day - basically there's a risk that you're expecting a `Vertex<DoubleTensor>` but you inadvertently get back a `Vertex<Double>`. Then you might set its value to be a Double, then when you try to get the value you get a `ClassCastException`.

Enforcing it to return a tensor vertex avoids this, although perhaps there are better ways to fix the problem.

I'm struggling to capture this in an example right now, but here's the example I sent round (but it's slightly different as it doesn't use VertexDictionary):

    @Test
    public void goodObserve() {
        DoubleVertex vertex = new GaussianVertex(0., 1.);
        vertex.observe(1.);
        DoubleTensor value = vertex.getValue();
        assertThat(value, equalTo(DoubleTensor.scalar(1.)));
    }

    @Test
    public void badObserve() {
        DoubleVertex vertex = new GaussianVertex(0., 1.);
        List<Vertex> vertices = ImmutableList.of(vertex);
        vertices.get(0).observe(1.);
        DoubleTensor value = vertex.getValue();
        assertThat(value, equalTo(DoubleTensor.scalar(1.)));
    }